### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by judicaelandria** [tldraw/tldraw-beta#50]
The beta website is kinda slow and the spinner is not loading, it's kinda stuck
more info in the video

https://user-images.githubusercontent.com/46365844/216378032-ab291e85-1c0c-4b2c-a9e8-671df7f2dbb6.mov

**Comment by judicaelandria** [comment]
Hey yes it is! and somehow I cannot paste an image in the multiplayer room, it just past the image name

**Comment by TodePond** [comment]
Could you share the specific image here possibly? :)

**Comment by judicaelandria** [comment]
<img width="1670" alt="Screenshot 2023-01-30 at 8 59 25 AM" src="https://user-images.githubusercontent.com/46365844/216997344-b3dbef3f-d6ea-441e-93ff-fdaeb8fafa67.png">

NB: I think the upload is working but it takes like 10+ to upload the image then it render it but after it rerender the image

**Comment by TodePond** [comment]
Hey we're now scaling down large images! Does it work better on https://staging.tldraw.com now?

**Comment by judicaelandria** [comment]
it is faster on local yes, but not on multiplayer mode

**Comment by judicaelandria** [comment]
it is faster on local yes, but not on multiplayer mode

**Comment by TodePond** [comment]
I'm thinking that might be due to the time it takes to download such a large file 🤔

**Comment by judicaelandria** [comment]
the non-multiplayer mode is downloading the large file too but it seems to be a lot faster than the multiplayer mode

**Comment by TodePond** [comment]
In non-multiplayer, the file is stored locally so it doesn't need to be downloaded - maybe that's it!

**Comment by judicaelandria** [comment]
yeah, maybe that's it! but I think it's better to show some sort of a loading state, just to tell the user that it's loading the image?

**Comment by TodePond** [comment]
Ok cool, glad we're figuring this out :) Yes I agree!
